src/Spago/Command/Fetch.purs

```diff
@@ -2,10 +2,12 @@ module Spago.Command.Fetch
   ( FetchEnv
   , FetchEnvRow
   , FetchOpts
+  , PackageDependencyInfo
```

Comment:
AFAIK, the behavior is unchanged. There are only 2 possible states that can occur when producing the `transitiveDeps` value here:
1. a package was selected (whether it was the root package or some other package is irrelevant). In which case, we get dependencies only for that package.
2. no package was selected. In which case, we get dependencies for all packages.

But, I updated this code so that I am also recording which package corresponds with which `PackageMap` value when calling `getTransitiveDeps`. Otherwise, the `Build.getBuildGlobs`' `dependencies` argument includes a given package's dependencies plus the other workspace package dependencies. Using `selected.package.dependencies` in `Build.getBuildGlobs` returns only the direct dependencies of that package, not the transitive ones.

Reply:
I see - then I think it's hard for me to see the need for both fields in this record:
- if we have a map of dependencies for each package, then we can just take a union of all the values when we need to have all the dependencies, that sounds like we don't need to keep the `dependencies` field around?

Reply:
I originally kept it because it led to less code churn due to the renaming of the field and its corresponding type while I worked on other things. Moreover, calculating `dependencies` here once meant I didn't need to recalculate it elsewhere in every usage thereafter. But, on the other hand, that means two pieces of state that could diverge at some point. So, I've dropped the `dependencies` field.